xlators/cluster/dht/src/dht-hashfn.c

```diff
@@ -51,20 +51,16 @@ dht_munge_name(const char *original, char *modified, size_t len, regex_t *re)
 
     ret = regexec(re, original, 2, matches, 0);
 
-    if (ret != REG_NOMATCH) {
+    if (caa_unlikely(ret == 0)) {
         if (matches[1].rm_so != -1) {
             new_len = matches[1].rm_eo - matches[1].rm_so;
-            /* Equal would fail due to the NUL at the end. */
-            if (new_len < len) {
+            if (new_len <= len) {
                 memcpy(modified, original + matches[1].rm_so, new_len);
-                modified[new_len] = '\0';
-                return new_len + 1; /* +1 for the terminating NULL */
+                return new_len;
             }
         }
     }
 
-    /* This is guaranteed safe because of how the dest was allocated. */
-    strcpy(modified, original);
```

Comment:
I'd be happy for review on this one. It wasn't clear to me why it's needed - if we return 0, we are ignoring essentially the modified name.